### PR TITLE
[NOD-1164] Remove the singleton from dbaccess, to enable multiple db connections in same run

### PR DESCRIPTION
**Pull request by svarogg**

### CODE REVIEW

File: blockdag/dag.go [hunk 1]
```diff
@@ -162,6 +163,103 @@ type BlockDAG struct {
 	startTime                       mstime.Time
 }
 
+// New returns a BlockDAG instance using the provided configuration details.
+func New(config *Config) (*BlockDAG, error) {
+	// Enforce required config fields.
```
Comment (stasatdaglabs) [inline blockdag/dag.go:168]: I would argue that a databaseContext is also required. Please a nil check here. 
Alternatively, remove all the nil checks and let it panic.

File: dbaccess/context.go [hunk 2]
```diff
@@ -11,24 +11,28 @@ type Context interface {
...
-func (*noTxContext) accessor() (database.DataAccessor, error) {
-	return db()
+type noTxContext struct {
+	backend *DatabaseContext
 }
 
-// NoTx creates and returns an instance of dbaccess.Context without an attached database transaction
-func NoTx() Context {
-	return noTxContextSingleton
+func (ctx *noTxContext) accessor() (database.DataAccessor, error) {
```
Comment (someone235) [inline dbaccess/context.go:18]: Do we maybe want to get rid of this function and implement it directly as part of `DatabaseContext`?

File: addrmgr/addrmanager.go [hunk 3]
```diff
@@ -58,6 +59,22 @@ type AddrManager struct {
 	localSubnetworkID  *subnetworkid.SubnetworkID
 }
 
+// New returns a new Kaspa address manager.
+// Use Start to begin processing asynchronous address updates.
```
[the remark this answers is not in the document]
Comment (svarogg) [reply addrmgr/addrmanager.go:63]: Why incorrect?

File: dbaccess/context.go (hunk 2 above)
Comment (svarogg) [reply dbaccess/context.go:18]: Nope, it makes the API of noTxContext and TxContext similar, allowing the rest of dbaccess function to be agonstic to tx/noTx

**Pull request merged**